src/System.Management.Automation/engine/hostifaces/Command.cs

```diff
@@ -109,6 +109,7 @@ internal Command(Command command)
             MergeToResult = command.MergeToResult;
             _mergeUnclaimedPreviousCommandResults = command._mergeUnclaimedPreviousCommandResults;
             IsEndOfStatement = command.IsEndOfStatement;
+            CommandInfo = command.CommandInfo;
```

Comment:
Should we clone `command.CommandInfo` too?

Reply:
I didn't see a clone method... But I suppose I could recreate it via the ctor.

Reply:
Yes, my thoughts was about ctor like for parameters below.

Reply:
The only problem is that `CommandInfo` is an abstract class and there are 9 implementations:
![image](https://user-images.githubusercontent.com/2644648/79139053-761f7580-7d6a-11ea-8f98-68de8ff14d39.png)

 of this abstract class and not all of them implement the clone ctor - 4 of them do:
![image](https://user-images.githubusercontent.com/2644648/79139108-8d5e6300-7d6a-11ea-8a8e-23d28cd463fa.png)

I could pattern match based on those types ... or implement a `.Clone()` on each of them...

Reply:
@TylerLeonhardt There is `CommandInfo.CreateGetCommandCopy`, but the result will have `CommandInfo.IsGetCommandCopy` set to `true`.  The main (only?) thing this does is adds a thread check around parameter metadata generation.  If an execution context is set, and parameter info is queried on a thread other than the pipeline thread, eventing is used to get back to the pipeline thread.

If someone is explicitly passing a non-GetCommandCopy, it shouldn't be tagged as one imo.  In other words, I don't think the `CommandInfo` should be cloned here.

Reply:
Yeah the `CommandInfo.CreateGetCommandCopy` method isn't implemented on all of the implemented types either so that might not be worth messing with.

I do have changes local that add an abstract method (`Clone()`) to `CommandInfo`. So I can include that in this PR if we wanna go that route. The changes weren't too heavy.

Reply:
I don't think it's necessary since `CommandInfo`s are read only to the consumer.

Reply:
I have a concern because we change Engine and we can catch tricky side effects in other scenarios.
I thing we need more reviewers. /cc @SteveL-MSFT @daxian-dbw @rjmholt